barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -220,6 +222,13 @@ void ClientIVC::accumulate(ClientCircuit& circuit,
                            const std::shared_ptr<MegaVerificationKey>& precomputed_vk,
                            const bool mock_vk)
 {
+    if (num_circuits > 0) {
```

Comment:
It seems better to just BB_ASSERT num_circuits > 0 with your error message then just always decrement. We're going to turn most asserts like this into runtime errors soon anyway